capa/rules/__init__.py

```diff
@@ -799,16 +850,19 @@ def from_dict(cls, d: Dict[str, Any], definition: str) -> "Rule":
         if isinstance(statements[0], ceng.Subscope):
             raise InvalidRule("top level statement may not be a subscope")
 
-        if scope not in SUPPORTED_FEATURES.keys():
-            raise InvalidRule("{:s} is not a supported scope".format(scope))
-
         meta = d["rule"]["meta"]
         if not isinstance(meta.get("att&ck", []), list):
             raise InvalidRule("ATT&CK mapping must be a list")
         if not isinstance(meta.get("mbc", []), list):
             raise InvalidRule("MBC mapping must be a list")
 
-        return cls(name, scope, build_statements(statements[0], scope), meta, definition)
+        # if the two statements are not the same, an InvalidRule() exception will be thrown
+        if scopes.static:
+            statement = build_statements(statements[0], scopes.static)
+        if scopes.dynamic:
+            # check if the statement is valid for the dynamic scope
+            _ = build_statements(statements[0], scopes.dynamic)
```

Comment:
it looks like `statement` is unbound here if there is no static scope. how can we better handle that scenario?

Reply:
yeah I have been operating under the assumption that rule authors must specify both the static and dynamic scopes (it gets checked in the `Scopes.form_dict()` method), and that all the features contained within each rule must be valid for both scopes.

Should I modify this assumption and make it possible to have just one scope defined per rule?

Reply:
ok, if both scopes must be present, that's fine. 

how are the results of the dynamic build statements ever used? what is the point of invoking that routine if the results are thrown away?

Reply:
> ok, if both scopes must be present, that's fine. maybe then we don't need to check if the property is present before referencing it?

ah true, my bad, that was a remnant of previous tests (allowing `Scopes` to have just one scope).

> how are the results of the dynamic build statements ever used? what is the point of invoking that routine if the results are thrown away?

`build_statements(statements[0], scopes.dynamic)` is called to make sure `statements[0]` is valid for the dynamic scope as well, since if it isn't then an _InvalidRule_ exception will be thrown.

> if it does do different things, then it seems that the results need to be saved off in different places for each scope?

yeah this is indeed the case, and for the time being I am not saving that anywhere since the statements are the same. I will create an issue for us to discuss how we want to move forward with building the statements, thus far I have had in mind only the scenario in which we'd create just one large statement tree that contains features that are valid for either the static scope or the dynamic one, and upon matching for a said scope the non-valid features would be just ignored; but I guess building two separate statements is a possibility as well, if that's the case then we'd need to save both statements indeed. I think we'll see what to do once we've decided on the previous issue (one large statement vs 2 separate statements).

Reply:
ok, sounds good. please add a comment explaining why these aren't used, and then lets come back to it as you indicated.